models.py

```diff
@@ -584,8 +607,27 @@ def forward(self, signal: shaping.BT, mask: shaping.BT = None, **kwargs) -> shap
 
 		power_spectrum = real_squared + imag_squared
 		log_mel_features = self.mel(power_spectrum).log()
+
+		'''
+		if self.satisfy_features_divisibility_by_32 and log_mel_features.shape[-1] % 32 != 0:
+			pad = int(math.ceil(log_mel_features.shape[-1] / 32) * 32) - log_mel_features.shape[-1]
+			log_mel_features = F.pad(log_mel_features, (0, pad), mode='constant', value = 0)
+		'''
+		if self.check_features_divisibility_by_32:
+			if signal.shape[-1] % (32 / 2) != 0:
+				warnings.warn('Shape of input signal not divisible by 16')
+
+			if log_mel_features.shape[-1] % 32 != 0:
+				warnings.warn('Shape of frontend output signal not divisible by 32')
+
 		return log_mel_features
 
+	@staticmethod
```

Comment:
please move this method in the other PR. the idea is to merge in just the output shape optimization, without modified padding behavior

Reply:
done